Makefile

```diff
@@ -1,22 +1,31 @@
 DOXYGEN := doxygen
 
 CXXFLAGS := -Wall -Wextra -Werror -Wfatal-errors
+CXXFLAGS := $(CXXFLAGS) -O3 -mtune=native -march=native
 CXXFLAGS := $(CXXFLAGS) -I $(CURDIR)/include 
 CXXFLAGS := $(CXXFLAGS) -std=c++14
 
 VPATH += include/fast_bvh
 
 headers := include/FastBVH.h \
            include/FastBVH/BBox.h \
-           include/FastBVH/BuildStrategy1.h \
            include/FastBVH/BuildStrategy.h \
            include/FastBVH/BVH.h \
-           include/FastBVH/Config.h \
+           include/FastBVH/impl/Config.h \
            include/FastBVH/Intersection.h \
            include/FastBVH/Iterable.h \
            include/FastBVH/Ray.h \
            include/FastBVH/Traverser.h \
-           include/FastBVH/Vector3.h
+           include/FastBVH/Vector3.h \
```

Comment:
Some headers were moved into `impl` so users aren't overwhelmed at looking through headers they're never going to directly use.